File: .tmp_py4013/batch_09_dump/2964_number_of_divisible_triplet_sums.js
// LeetCode 2964 - Number of Divisible Triplet Sums
// https://leetcode.com/problems/number-of-divisible-triplet-sums/

var divisibleTripletCount = function(nums, d) {
    const n = nums.length;
    let ans = 0;
    for (let i = 0; i < n; i++) {
        const freq = new Map();
        for (let j = i + 1; j < n; j++) {
            const need = (d - (nums[i] + nums[j]) % d) % d;
            ans += freq.get(need) || 0;
            const key = nums[j] % d;
            freq.set(key, (freq.get(key) || 0) + 1);
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check13/2964_number_of_divisible_triplet_sums.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 2964 - Number of Divisible Triplet Sums
// https://leetcode.com/problems/number-of-divisible-triplet-sums/

use std::collections::HashMap;

impl Solution {
    pub fn divisible_triplet_count(nums: Vec<i32>, d: i32) -> i32 {
        let n = nums.len();
        let mut ans = 0;
        for i in 0..n {
            let mut freq: HashMap<i32, i32> = HashMap::new();
            for j in (i + 1)..n {
                let need = (d - (nums[i] + nums[j]) % d) % d;
                ans += *freq.get(&need).unwrap_or(&0);
                *freq.entry(nums[j] % d).or_insert(0) += 1;
            }
        }
        ans
    }
}
